core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -1896,6 +1896,11 @@ class KafkaApis(val requestChannel: RequestChannel,
               .setTopicConfigErrorCode(Errors.NONE.code)
           }
         }
+        val topicIds = zkClient.getTopicIdsForTopics(results.asScala.map(result => result.name()).toSet)
```

Comment:
If going to ZK here is too slow, another option is to provide a callback to adminManager.createTopics which can be called after createTopicWithAssignment. The idea is that createTopicWithAssignment (and writeTopicPartitionAssignment) would return the topic ID to avoid an extra call to ZK. I wasn't sure which option was better.

Reply:
I think we can remove support for zk when using topic-id since KIP-500 will remove zk, also ping @rajinisivaram to have a look.

Reply:
Currently the KIP-516 work relies on ZK for storing the topic IDs. I think we will need to make some more changes when KIP-500 fully removes ZK

Reply:
Here could we use controllerContext since createTopicRep is handled at the active controller.